hera_cal/tests/test_frf.py

```diff
@@ -1066,3 +1067,241 @@ def test_tophat_linear_argparser(self):
+    frop = frf.get_frop_for_noise(times, filt_cent, filt_hw, 
+                                  freqs=data.freqs[freq_slice], 
+                                  weights=weights[cross_antpairpol][:, freq_slice], 
+                                  coherent_avg=True, cutoff=eval_cutoff,
+                                  dlst=dlst, nsamples=nsamples[cross_antpairpol][:, freq_slice], 
+                                  t_avg=300., bl_vec=bl_vec)
+    
+    cov = frf.get_FRF_cov(frop, var_for_frop)
+
+    # Check that at least it's (close to) hermitian at every frequency
+    # Tests for value sensibility exist in single baseline PS notebook
+    for freq_ind in range(1024 - 37):
+        assert np.allclose(cov[freq_ind], cov[freq_ind].T.conj())
+
+
+    # Pretend we forgot to slice things
+    with pytest.raises(ValueError, match="nsamples has wrong shape"):
+            frop = frf.get_frop_for_noise(times, filt_cent, filt_hw, 
+                                          freqs=data.freqs[freq_slice], 
+                                          weights=weights[cross_antpairpol][:, freq_slice], 
+                                          coherent_avg=True, cutoff=eval_cutoff,
+                                          dlst=dlst, nsamples=nsamples[cross_antpairpol], 
+                                          t_avg=300., bl_vec=bl_vec)
+            
+    with pytest.raises(ValueError, match="weights has wrong shape"):
+            frop = frf.get_frop_for_noise(times, filt_cent, filt_hw, 
+                                          freqs=data.freqs[freq_slice], 
+                                          weights=weights[cross_antpairpol], 
+                                          coherent_avg=True, cutoff=eval_cutoff,
+                                          dlst=dlst, nsamples=nsamples[cross_antpairpol], 
+                                          t_avg=300., bl_vec=bl_vec)
+
+def test_get_corr_and_factor():
+    # Actually just going to test an easy analytic case here
+    base = np.array([[2, 1, 0], 
+                     [1, 3, 1], 
+                     [0, 1, 4]])
+    cov = np.array([base, 2 * base])
+    corr = frf.get_corr(cov)
+
+    answer = np.array([[1, 1/np.sqrt(6), 0], 
+                       [1/np.sqrt(6), 1, 1/np.sqrt(12)],
+                       [0, 1/np.sqrt(12), 1]])
+
+    answer = np.array([answer, answer])
+
+    assert np.allclose(corr, answer)
+
+    factor = frf.get_correction_factor_from_cov(cov)
+
+    # A block diagonal "implementation"
+    blocklen = np.prod(cov.shape[:2])
+    sum_sq = 7
+    Neff = blocklen**2 / sum_sq
+    factor_answer = blocklen / Neff
+
+    assert np.allclose(factor, factor_answer)
+
+    tslc = slice(1, 3)
+    factor_slice = frf.get_correction_factor_from_cov(cov, tslc=tslc)
+
+    blocklen = 4
+    sum_sq = 2 * (2 + 2 / 12)
+    Neff = blocklen**2 / sum_sq
+    factor_slice_answer = blocklen / Neff
+
+    assert np.allclose(factor_slice, factor_slice_answer)
+
+
+    
+
+
+
+
+    
+
+
```

Comment:
don't think you meant to add all this whitespace

Reply:
Correct -- rampant pressing of the enter key. I've removed it.